### PR TITLE
Bump cuda version and drop cuda version tags for `ubuntu:jammy` based images

### LINKED ISSUE #282: Swich the base image to Ubuntu 22.04 LTS
Just as the release dates of R4.0.0 and Ubuntu 20.04 were only a few days apart, the release date of R4.2.0 could be right after Ubuntu 22.04.
Immediately after the release of jammy, there is a possibility that RSPM binary installation, RStudio, etc. are not compatible with jammy, so we may need to modify the build system.

https://discourse.ubuntu.com/t/jammy-jellyfish-release-schedule/23906

```shell
$ docker run --rm -it rocker/r-ver Rscript -e "read.csv('/usr/share/distro-info/ubuntu.csv') |> tail(1)"
     version        codename series    created    release        eol eol.server
36 22.04 LTS Jammy Jellyfish  jammy 2021-10-14 2022-04-21 2027-04-21 2027-04-21
      eol.esm
36 2032-04-21
```

### LINKED ISSUE #503: Cuda versioning
- Remove `cuda10.1` images (R 4.2.2 or latetr?)
- Switch cuda11.1.1 to cuda 11.7.0 (R 4.2.2 or latetr)

Related to #282

### PR DESCRIPTION
**Pull request by eitsupi**
close #282, close #503

We have maintained two `rocker/cuda` variants, cuda10.1 and cuda11.1, but now that the base image will be `ubuntu:jammy`, we will unify the cuda variants into one, as we can no longer use anything other than cuda11.7.X.

1. cuda version tags, like `rocker/cuda:cuda10.1`, `rocker/cuda:4.2.1-cuda10.1` will be obsolete for R 4.2.2 or later images.
2. The aliases of `rocker/cuda` which was pushed to `rocker/r-ver` (`rocker/r-ver:X.Y.Z-cuda10.1` and `rocker/r-ver:X.Y.Z-cuda11.1`) will be obsolete.

Only the R version tags like `rocker/cuda:4.2.2`, `rocker/cuda:latest` will remain.

Note: This Pull Request can only be merged after the R 4.2.2 release.

**ToDo**

- [x] Update `stacks/4.2.1.json` by hand.
  - [x] Remove `latest` tags.
  - [x] Change CRAN and CTAN urls.
- [x] Run `./build/make-stacks.R` -> `make clean` -> `make setup`
- [x] Update NEWS.md
- [ ] Update the reference page

### CODE REVIEW

**Review by cboettig — APPROVED** [review]
nice, looks good!

File: build/make-stacks.R [hunk 1]
```diff
@@ -110,7 +110,7 @@ library(gert)
   return(.url)
 }
 
-.cuda_baseimage_tag <- function(ubuntu_series, other_variants = "11.1.1-cudnn8-devel") {
+.cuda_baseimage_tag <- function(ubuntu_series, other_variants = "11.7.1-cudnn8-devel") {
```
Comment (eitsupi) [inline build/make-stacks.R:113]: It may be worthwhile to automatically update the cuda version instead of hard-coding it.

**Comment by eitsupi** [comment]
I did a manual update as planned, so I'll merge this.

**Comment by cboettig** [comment]
awesome!

**Pull request merged**